### ISSUE TITLE
Add new applicable_kinds to distinguish between parameter of regular functions and parameters of anonymous functions

### ISSUE THREAD

**Issue by patelmitul** [dotnet/roslyn#70037]
**Brief description:**

We have few [applicable_kinds](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/naming-rules#propertyname-and-propertyvalue) available to define naming conversions for different kind of object. Right now all parameters are grouped in  one kind "parameter". Can we add new kind to distinguish between parameter of regular functions and parameters of anonymous functions?

**Languages applicable:**

Both

**Code example that the analyzer should report:**

```
// Regular function parameter: Require lower case 'p' as prefix and capitalization as pascal_case
// Anonymous function parameter: Does not require prefix and capitalization as camel_case
public int GetCount(IList<int> pMyItems) {
    return pMyItems.Where((x) => x > 0).Count();
}
```

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).
